docs/angular/build-options.md

```diff
@@ -0,0 +1,252 @@
+
+```diff
+import { enableProdMode, importProvidersFrom } from '@angular/core';
+import { bootstrapApplication } from '@angular/platform-browser';
+import { RouteReuseStrategy, provideRouter } from '@angular/router';
+- import { IonicModule, IonicRouteStrategy } from '@ionic/angular';
++ import { provideIonicAngular, IonicRouteStrategy } from '@ionic/angular/standalone';
+
+import { routes } from './app/app.routes';
+import { AppComponent } from './app/app.component';
+import { environment } from './environments/environment';
+
+if (environment.production) {
+  enableProdMode();
+}
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    { provide: RouteReuseStrategy, useClass: IonicRouteStrategy },
+-   importProvidersFrom(IonicModule.forRoot({ mode: 'md' })),
++   provideIonicAngular({ mode: 'md' }),
+    provideRouter(routes),
+  ],
+});
+```
+
+4. Remove any references to `IonicModule` found elsewhere in your application.
+
+5. Update any existing imports from `@ionic/angular` to import from `@ionic/angular/standalone` instead.
+
+```diff
+- import { Platform } from '@ionic/angular';
++ import { Platform } from '@ionic/angular/standalone';
+```
+
+6. Add imports for each Ionic component in the Angular component where they are used. Be sure to pass the imports to the `imports` array on your Angular component.
+
+```diff
+import { Component } from '@angular/core';
++ import { IonApp, IonRouterOutlet } from '@ionic/angular/standalone';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.scss'],
+  standalone: true,
++ imports: [IonApp, IonRouterOutlet],
+})
+export class AppComponent {
+  constructor() {}
+}
+```
+
+7. If you are using Ionicons, define the icon SVG data used in each Angular component using `addIcons`. This allows you to continue referencing icons by string name in your component template. Note that you will need to do this for any additional icons added.
+
+```diff
+import { Component } from '@angular/core';
++ import { IonIcon } from '@ionic/angular/standalone';
++ import { addIcons } from 'ionicons';
++ import { alarm, logoIonic } from 'ionicons/icons';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.scss'],
+  standalone: true,
++ imports: [IonIcon],
+})
+export class TestComponent {
+  constructor() {
+    addIcons({ alarm, logoIonic });
+  }
+}
+```
+
+8. Remove the following code from your `angular.json` file if present. Note that it may appear multiple times.
+
+```diff
+- {
+-   "glob": "**/*.svg",
+-   "input": "node_modules/ionicons/dist/ionicons/svg",
+-   "output": "./svg"
+- }
+```
+
+### Limitations
+
+1. Your application must already make use of standalone APIs. Using Ionic UI components as standalone components in an application that uses `NgModule` is not supported.
```

Comment:
Does this mean you can't be doing either of these two (following) things in your app? The Angular docs make it sound like it's easy to gradually migrate to standalone components. Are we saying that's not possible when using Ionic standalone components?

1. [Using existing NgModules in a standalone component](https://angular.io/guide/standalone-components#using-existing-ngmodules-in-a-standalone-component)
2. [Using standalone components in NgModule-based applications](https://angular.io/guide/standalone-components#using-standalone-components-in-ngmodule-based-applications)

If neither of those approaches is possible, I think we should mention it in Standalone > Overview > Drawbacks as well as in the migration section.

Reply:
Actually I think I may have gotten some of this wrong. Let me do some digging and then I'll re-request you for review

Reply:
I actually didn't know you could use standalone components in an NgModule-based application, so thanks for pointing that out! I updated the migration guide to have two guides: 1 for standalone-based apps and 1 for NgModule-based apps.

I removed the limitation that said you needed to have the standalone architecture. Also, I moved the "all at once" limitation to the beginning of the migration guide so it's clear you need to migrate Ionic UI components all at once.

See https://github.com/ionic-team/ionic-docs/pull/3115/commits/be2918d1072aa4291a9a3162bcc2b6ba14379ba4

Will re-request review when preview website is deployed.

Reply:
I tried it out with an NgModule app, and the instructions were clear and had everything i needed!